### PR TITLE
Test/jax core trainer unit test

### LINKED ISSUE #550: [TEST] Jax System Core Trainer
### What do you want to test?
Jax system core trainer (in `mava/systems/jax/`)

### Outline of test structure
* Unit tests
* Test components and hooks

### Definition of done
Passing checks, cover all hooks, edge cases considered

### Mandatory checklist before making a PR
* [ ] The success criteria laid down in “Definition of done” are met.
* [ ] Test code is documented - docstrings for methods and classes, static types for arguments.
* [ ] Documentation is updated - README, CONTRIBUTING, or other documentation.

### PR DESCRIPTION
**Pull request by RuanJohn**
## What?
Added unit tests for Jax core trainer. 
## Why?
Part of ongoing efforts to up test code coverage for Mava callback redesign. 
## How?
Added test. 
## Extra
Closes #550

### CODE REVIEW

File: tests/jax/core_system_components/trainer_test.py [hunk 1]
```diff
@@ -13,4 +13,57 @@
...
+
+from mava.callbacks import Callback
+from mava.systems.jax.trainer import Trainer
+from tests.jax.hook_order_tracking import HookOrderTracking
+
+
+class TestTrainer(HookOrderTracking, Trainer):
+    def __init__(
+        self,
+        config: SimpleNamespace,
```
Comment (mmorris44) [inline tests/jax/core_system_components/trainer_test.py:29]: At some point, we still need to go rename a bunch of these, since in a bunch of the core components, they call this `config` instead of `store`. Will make a separate PR for this though.

**Review by mmorris44 — DISMISSED** [review]
Nice concise tests, thanks @RuanJohn

**Review by OmaymaMahjoub — APPROVED** [review]
Looks good. Thank you @RuanJohn just a small thing that may be unnecessary if you can add a comment up the file describing what the file is about!

**Comment by RuanJohn** [comment]
Added a comment back in @OmaymaMahjoub :call_me_hand:

**Review by mmorris44 — APPROVED** [review]
